README.md

```diff
@@ -75,364 +70,360 @@ Please note that using Ollama requires a large (>4GB) download of the docker ima
-    ```
+    Then, we'll load the data from the huggingface dataset using the `load_wiki_articles` function:
     
-    Related documentation: [load dataset from huggingface](/docs/utils/load_dataset_from_huggingface.md).
-
+    <details>
+    <summary>Click to see the python code for `load_wiki_articles`</summary>
+    
+    ```python
+    async def load_wiki_articles():
+        # to keep the demo fast, we have some simple limits
+        num_articles = 10
+        max_text_length = 1000
+        
+        wiki_dataset = load_dataset("wikimedia/wikipedia", "20231101.en", split=f"train", streaming=True)
+        
+        async with pool.connection() as conn:
+            async with conn.cursor() as cur:
+                for article in wiki_dataset.take(num_articles):
+                    await cur.execute(
+                        "INSERT INTO wiki (url, title, text) VALUES (%s, %s, %s)",
+                        (article['url'], article['title'], article['text'][:max_text_length])
+                    )
+                await conn.commit()
+    ```
+    </details>
 1. **Create a vectorizer for `wiki`**
 
-    To enable semantic search on the `wiki` table, we need to create vector embeddings for the `text` column.
+    To enable semantic search on the `text` column of the `wiki` table, we need to create vector embeddings for that column.
     We use a vectorizer to automatically create these embeddings and keep them in sync with the data in the  `wiki` table.
+    We do this in the `create_vectorizer` function:
     
-    ```sql
-    SELECT ai.create_vectorizer(
-         'wiki'::regclass,
-         loading => ai.loading_column('text'),
-         destination => 'wiki_embeddings',
-         embedding => ai.embedding_ollama('all-minilm', 384)
-    );
-    ```
+    <details>
+    <summary>Click to see the python code for `create_vectorizer`</summary>
     
-     Related documentation: [vectorizer usage guide](/docs/vectorizer/overview.md) and [vectorizer API reference](/docs/vectorizer/api-reference.md).
+    ```python
+    async def create_vectorizer():
+        vectorizer_statement = CreateVectorizer(
+            source="wiki",
+            target_table='wiki_embedding_storage',
+            loading=LoadingColumnConfig(column_name='text'),
+            embedding=EmbeddingOllamaConfig(model='all-minilm', dimensions=384, base_url="http://localhost:11434")
+        ).to_sql()
+        
+        try:
+            async with pool.connection() as conn:
+                async with conn.cursor() as cur:
+                    await cur.execute(vectorizer_statement)
+                await conn.commit()
+        except Exception as e:
+            if "already exists" in str(e):
+                # ignore if the vectorizer already exists
+                pass
+            else:
+                raise e
+    ```
+    </details>
+    To learn more about vectorizers, see the [vectorizer usage guide](/docs/vectorizer/overview.md).
```

Comment:
This link doesn't seem to render correctly for me, but maybe thats just a github error?